dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/cdi/FutureEMBuilder.java

```diff
@@ -437,4 +437,11 @@ public String toString() {
                         .append('#').append(module);
         return b.toString();
     }
+
+    // Ensures order of DDL generation based on repository interface name
+    @Override
+	@Trivial
+    public int compareTo(FutureEMBuilder o) {
```

Comment:
Did you consider that each FutureEMBuilder may actually support multiple repositoryInterface types?  This field is set to the first one reported by CDI.  For each repository interface, the code looks for an existing FutureEMBuilder that matches on datastore, app, module, and classloader.  See FutureEMBuilder.equals.

Perhaps FutureEMBuilder should be updated to keep track of all repository interfaces it supports, and then use the one that is alphabetically first?

Or perhaps CDI will always find them in the same order and it doesn't matter?